include/real/exact_number.hpp

```diff
@@ -693,190 +694,322 @@ namespace boost {
                 }
 
-                if (divisor == (*this)) { 
-                    (*this) = tmp;
-                    (*this).positive = positive;
+                if(divisor == (*this)){
+                    (*this) = one;
                     return;
                 }
 
-                if ((*this) == zero)
-                    return;
-                    
-                if (divisor == zero)
-                    throw(boost::real::divide_by_zero());
-
-                // N < D --> 0 < abs(Q) < 1
-                if (numerator < divisor) {
-                        left = exact_number<T>(); // 0
-                        right = tmp; // 1
-                    } else { // assuming D > 1. N > D ---> 1 < N / D < N
-                        left = tmp; // 1
-                        right = numerator;
-                    }
+                const T base = (std::numeric_limits<T>::max() / 4) * 2 - 1;
+                exact_number<T> half;
+                half.digits = {base / 2 + 1};
+
+                exact_number<T> left, right, numerator, denominator, residual, length;
+                numerator = (*this);
+                denominator = divisor;
+                numerator = numerator.abs();
+                denominator = denominator.abs();
 
-                // distance = (right - left) / 2
-                distance = (right - left) * half;
-                (*this) = left + distance;
-                // N/D = Q -> QD -N = 0
-                residual = (*this) * divisor - numerator;
-                if (residual == zero) {
-                    this->exponent += exponent_dif;
+                int exponent_diff = (this->exponent - 1) - (denominator.exponent - 1);
+                numerator.exponent = 1;
+                denominator.exponent = 1;
+                
+
+                if(numerator > denominator){
+                    left = one;
+                    right = numerator;
+                }else{
+                    left = zero;
+                    right = one;
+                }
+
+                length = (right - left) * half;
+                (*this) = length + left;
+
+                residual = (*this) * denominator - numerator;
+                if(residual == zero){
+                    this->exponent += exponent_diff;
                     this->positive = positive;
                     return;
                 }
 
-                // calculate the result
-                // continue the loop while we are still inaccurate (up to max precision)
-                while ((residual.abs() > min_boundary_p) && 
-                       (distance.exponent >= min_boundary_p.exponent)) {
-                    // result too small, try halfway between left and (*this) 
-                    if (residual < min_boundary_n) {
+                exact_number<T> maximum_error, neg_maximum_error;          // +-epsilon error in quotient
+                maximum_error.digits = {1};
+                maximum_error.exponent = (-1)*maximum_precision;
```

Comment:
-maximum_precision; ?

Reply:
I will change that.

Reply:
Remember to not use the maximum_precision as the error. pass a max_error argument and use that max error to ensure you approximated binary_search_division will return a result with an error smaller or equal to max_error.

Reply:
Again, the confusion is due to the name, "maximum_precision" here is exponent of max_error, not max precision of iterator. I'll change the name